boosters/node-beginner/tests/07_exercise.test.js

```diff
@@ -0,0 +1,77 @@
+const request = require("supertest");
+const app = require("../src/server");
+const teardown = require("./config/teardown");
+const uuid = require("uuid/v4");
+const setup = require("./config/setup");
+
+const users = [
+  { firstName: "Henry", lastName: "Chao", id: uuid() },
+  { firstName: "Andrew", lastName: "Mayer", id: uuid() },
+  { firstName: "Liang", lastName: "Chen", id: uuid() },
+];
+
+const repos = users.reduce(
+  (acc, curr) => [
+    ...acc,
+    { name: "repo1", userId: curr.id, isPrivate: false, id: uuid() },
+    { name: "repo2", userId: curr.id, isPrivate: true, id: uuid() },
+  ],
+  []
+);
+
+describe("Not sure yet", () => {
+  beforeAll(async () => {
+    await setup("Users", users)();
+    return setup("Repos", repos)();
+  });
+
+  afterAll(teardown);
+
+  xit("Should return all public repositories and all repositories of current user", async () => {
+    const res = await Promise.all(
+      users.map(user =>
+        request(app)
+          .get("/repos/")
+          .send({ userId: user.id })
+      )
+    );
+    const test = res.reduce(
+      (acc, curr1, index) =>
+        acc &&
+        curr1.body.reduce(
+          (acc, curr) => acc && (users[index].id === curr.userId || !curr.isPrivate),
+          true
+        ),
+      true
+    );
+    expect(test).toEqual(true);
+  });
+
+  it("Should return a single repository if it's either public or owned by current user. Otherwise return null", async () => {
```

Comment:
Split this out into three test cases. One if it's public, another if it's owned by the current user, and one where the repo is not returned.